multicluster/test/e2e/main_test.go

```diff
@@ -83,9 +94,53 @@ func testMain(m *testing.M) int {
 	rand.Seed(time.Now().UnixNano())
 
 	ret := m.Run()
+	if ret != 0 {
+		return ret
+	}
+
+	log.Println("Start E2E test with WireGuard")
+	for _, clusterName := range testData.clusters {
+		if err := enableWireGuard(clusterName); err != nil {
+			log.Fatalf("Error when enabling WireGuard encryption, error: %v", err)
+		}
+	}
+
+	ret = m.Run()
 	return ret
 }
 
+func enableWireGuard(clusterName string) error {
+	data := testData.clusterTestDataMap[clusterName]
+	configMap, err := data.GetConfigMap(antreaAgentNamespace, antreaAgentConfigName)
+	if err != nil {
+		return err
+	}
+	antreaAgentConfig := &agentconfig.AgentConfig{}
+	if err := yaml.Unmarshal([]byte(configMap.Data[antreaAgentConfName]), antreaAgentConfig); err != nil {
+		return err
+	}
+	antreaAgentConfig.Multicluster.TrafficEncryptionMode = "wireGuard"
+	conf, err := yaml.Marshal(antreaAgentConfig)
+	if err != nil {
+		return err
+	}
+	configMap.Data[antreaAgentConfigName] = string(conf)
+	if err := data.UpdateConfigMap(configMap); err != nil {
+		return err
+	}
+	podList, err := data.ListPods(antreaAgentNamespace, antreaLabelSelector)
```

Comment:
I think controller is not impacted by the configMap change, you may add one more label filter "component=antrea-agent"?